contracts/extensions/UnitToken/UnitToken.sol

```diff
@@ -0,0 +1,84 @@
+pragma solidity ^0.8.0;
+
+// SPDX-License-Identifier: MIT
+
+import "../core/DaoConstants.sol";
+import "../core/DaoRegistry.sol";
+import "../extensions/bank/Bank.sol";
+import "../guards/AdapterGuard.sol";
+import "./interfaces/IConfiguration.sol";
+import "../adapters/interfaces/IVoting.sol";
+
+import "../../utils/IERC20.sol";
+import "@openzeppelin/contracts/utils/Address.sol";
+import "../../helpers/SafeERC20.sol";
+
+/**
+MIT License
+
+Copyright (c) 2020 Openlaw
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in all
+copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+ */
+
+
+/**
+ * 
+ The UnitTokenContract  is a contract to handle an erc20 token that 
+ mirrors the number of Units/voting weight held by DAO members. 
+
+
+  */
+contract UnitTokenContract is DaoConstants, AdapterGuard {
+	
+
+	constructor(address unitToken) public {
```

Comment:
So since this is an extension, the constructor should have two values
1. the DaoRegistry
2. the internal token address (usually unit but could be another internal token)

Then it should "simply" implement ERC20 interface where every transfer is a transfer by using the bank extension
and we should have a configuration in the dao registry to know whether the token is transferable or not

Reply:
for the transfers of the reserved address `UNITS` and `unitToken` I was going to use  
  `Bank.internalTransfer(from, to, unitToken, amount);` for both of them.  Instead of `IERC20.transferFrom(from, to, unitToken, amount);` for the `unitToken`.  WDYT?